packages/components/src/edit-in-place-control/index.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useRef, useEffect } from '@wordpress/element';
+import { ENTER, ESCAPE } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
+
+function EditInPlaceControl( { label = '', onClick = noop, onChange = noop } ) {
+	const [ edit, setEdit ] = useState( false );
+	const [ value, setValue ] = useState( label );
+	const prevValue = useRef();
```

Comment:
There's a hook `usePrevious` in `@wordpress/compose` 😉 .

Reply:
`usePrevious(value)` would update on every `value` change and here we need to update when `edit` changes

```
useEffect( () => {
		prevValue.current = value;
		if ( edit ) {
			inputRef.current.focus();
			inputRef.current.select();
		} else {
			buttonRef.current.focus();
		}
	}, [ edit ] );
```